### PR TITLE
:sparkles: feat: certify CAPG

### LINKED ISSUE #786: CAPG certification
### What would you like to be added (User Story)?

As a user I would like to be able to deploy clusters using the [infrastructure provider for GCP](https://github.com/kubernetes-sigs/cluster-api-provider-gcp).

### Detailed Description

Validate CAPG installation together with Turtles via `CAPIProvider` and test that cluster provisioning is feasible. Managed clusters is a experimental feature in CAPG so we should also ensure that we're able to deploy GKE clusters.

### Anything else you would like to add?

Additionally we could also validate ClusterClass provisioning and investigate what features we're missing from the provider.

### Label(s) to be applied

/kind feature

### PR DESCRIPTION
**Pull request by salasberryfin**
**What this PR does / why we need it**:

This certifies the Cluster API Provider GCP:

- Add CAPG to the list of providers in clusterctl `ConfigMap`.
- Configure credentials for GitHub Actions.
- Install CAPG in E2E test environment via `CAPIProvider`.
- Cover GCP cluster provisioning (GKE) in E2E test suites.

**Which issue(s) this PR fixes**:
Fixes #786 

**Special notes for your reviewer**:

CAPG cluster templates required you to specify the GCP Project ID. To keep this stored as a secret in GitHub, it cannot be used for substitution as we do with other environment variables (e.g. `WORKER_MACHINE_COUNT`) because it is not accessible when actions are triggered by PR webhooks. This is why a new field `IsGCPCluster` is added to the test input struct `CreateMgmtV3UsingGitOpsSpecInput` to only populate this variable from secret when it is explicitly requested.

Follow-up to this -> Create a fork of the provider and build our own images: https://github.com/rancher/highlander/issues/109

**Checklist**:

- [x] squashed commits into logical changes
- [ ] includes documentation
- [ ] adds unit tests
- [x] adds or updates e2e tests

### CODE REVIEW

File: internal/controllers/clusterctl/config.yaml
Comment (Danil-Grigorev) [inline internal/controllers/clusterctl/config.yaml:25]: Should this repo be included in forks build?

**Review by Danil-Grigorev — APPROVED** [review]
LGTM, pending e2e, just left some thoughts to look into later

File: internal/controllers/clusterctl/config.yaml
Comment (salasberryfin) [reply internal/controllers/clusterctl/config.yaml:25]: Yes, correct. I have this in my TODO list but I wanted to have it all working first and then I'll create a fork and build process like we do with other providers.

File: test/e2e/data/capi-operator/full-providers.yaml
Comment (salasberryfin) [reply test/e2e/data/capi-operator/full-providers.yaml:33]: Good point. I'll add this.

File: internal/controllers/clusterctl/config.yaml
Comment (salasberryfin) [reply internal/controllers/clusterctl/config.yaml:25]: Added a new issue to address this as a follow-up https://github.com/rancher/highlander/issues/109

File: test/e2e/suites/import-gitops-v3/import_gitops_v3_test.go
Comment (furkatgofurov7) [inline test/e2e/suites/import-gitops-v3/import_gitops_v3_test.go:161]: Non-blocking question: if I am correct, new additions to e2e is not tested (full test running only in nightly tests) in the current PR, have you tried it running maybe locally?

**Review by furkatgofurov7 — APPROVED** [review]
Thanks for working on this addition, LGTM!

File: test/e2e/suites/import-gitops-v3/import_gitops_v3_test.go
Comment (salasberryfin) [reply test/e2e/suites/import-gitops-v3/import_gitops_v3_test.go:161]: Yes, that is correct. There's no way to test this on PR tests as accessing secrets is not allowed for security reasons. It passes locally and I hope there's no issue when running in GitHub Actions, but I'll keep an eye on it after merging and try to spot any errors.

Comment (furkatgofurov7) [reply test/e2e/suites/import-gitops-v3/import_gitops_v3_test.go:161]: Thanks for confirming @salasberryfin.

File: internal/controllers/clusterctl/config.yaml
Comment (furkatgofurov7) [reply internal/controllers/clusterctl/config.yaml:25]: Should we wait until images from the fork are built (looks like they soon be available?) first and update the ref here, before taking this PR in?

Comment (furkatgofurov7) [reply internal/controllers/clusterctl/config.yaml]: could it be related to some kustomization where it misses image patching, i.e https://github.com/kubernetes-sigs/cluster-api/blob/main/Makefile#L1319-L1322?

**Review by furkatgofurov7 — APPROVED** [review]

File: internal/controllers/clusterctl/config.yaml
Comment (furkatgofurov7) [reply internal/controllers/clusterctl/config.yaml]: @salasberryfin eventually I believe the source of truth for image as I checked is:

` image: registry.k8s.io/cluster-api-gcp/cluster-api-gcp-controller:v1.8.0`

coming from https://github.com/rancher-sandbox/cluster-api-provider-gcp/blob/main/Makefile#L120-L121 and that ends up eventually without modification in the release artifact files

Comment (furkatgofurov7) [reply internal/controllers/clusterctl/config.yaml]: Alright, I found the root cause of it:

For CAPI manifests, it looks fine:

`image: registry.rancher.com/rancher/cluster-api-controller:v1.8.4`

because Makefile definition of it uses `?` mark in var: https://github.com/rancher-sandbox/cluster-api/blob/main/Makefile#L203C1-L203C45

resulting in image patching correctly, since we explicitly pass `PROD_REGISTRY` in release workflow: https://github.com/rancher/clusterapi-forks/blob/main/.github/workflows/publish_release/action.yaml#L51C9-L51C22


However, both [CAPZ](https://github.com/rancher-sandbox/cluster-api-provider-azure/blob/main/Makefile#L162)/[CAPG](https://github.com/rancher-sandbox/cluster-api-provider-gcp/blob/main/Makefile#L120) omits `?`, hence image is not overwritten as expected.

**Review by Danil-Grigorev — APPROVED** [review]

File: internal/controllers/clusterctl/config.yaml
Comment (furkatgofurov7) [reply internal/controllers/clusterctl/config.yaml]: We will tackle possibility of image overriding in upstream CAPZ/CAPG repos.

**Pull request merged**
